external-crates/move/crates/move-compiler/tests/move_2024/parser/mut_field_pun_invalid_pack.move

```diff
@@ -1,7 +1,7 @@
 module a::m {
     public struct S { f: u64 }
 
-    public fun foo(): S {
```

Comment:
Why did we lose the return type?

Reply:
In this particular case, I'd expect this to parse to code that works out

Reply:
And it does (brought the return type back)